Client/Pages/Transfer.razor

```diff
@@ -0,0 +1,155 @@
+            <h5 class="card-title">Wallet Informations :</h5>
+            <h6 class="card-subtitle mb-2 text-muted">Amount : @wallet.Amount</h6>
+            <h6 class="card-subtitle mb-2 text-muted">Currency : @wallet.Currency</h6>
+            <br />
+            <h5 class="card-title">Make a transfer to :</h5>
+            <div class="row">
+                <div class="col-6">
+                    <label>Recipient's name :</label>
+                    <input type="email" class="form-control" placeholder="Recipient's name..." @bind="data.Username">
+                    @if (StatusData(1,data.Username))
+                    {
+                        <small class="form-text text-success">Recipient's name is valid.</small>
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">Recipient's name is invalid.</small>
+                    }
+                </div>
+                <div class="col-6">
+                    <label>Recipient's Wallet Id :</label>
+                    <input class="form-control" placeholder="Recipient's Wallet Id..." @bind="targetId">
+                    @if (StatusData(2,targetId))
+                    {
+                        <small class="form-text text-success">The Wallet Id format is valid.</small>
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">The Wallet Id format is invalid.</small>
+                    }
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-6 p-2">
+                    <label>Amount :</label>
+                    <input type="text" class="form-control" placeholder="The Amount..." @bind="Amount">
+                    @if (StatusData(3,Amount))
+                    {
+                        <small class="form-text text-success">The amount entered is valid.</small>
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">The amount entered is invalid.</small>
+                    }
+                </div>
+                <div class="col-6 p-3">
+                    <br />
+                    @if (StatusData(1,data.Username) && StatusData(2,targetId) && StatusData(3,Amount))
+                    {
+                        <button class="btn btn-danger" @onclick=MakeTransfer>Finish the transfer</button>
+                        <small class="form-text text-success">The transfer can be done now.</small>
+                    }
+                    else
+                    {
+                        <button class="btn btn-danger">Finish the transfer</button>
+                        <small class="form-text text-danger">The transfer can't be done now.</small>
+                    }
+                </div>
+            </div>
+
+        </div>
+    </div>
+    <br />
+    @if (!Status)
```

Comment:
That's good, but I suggest looking at other possible scenarios. The status could be false for numerous reasons, including the database not being available (because someone is pulling one million users out of it right now ;) ), or some random null reference exception which wasn't properly handled, or literally anything else.
In this situation, for a proper user experience, I recommend turning "status" into an enum, and having multiple possible error cases, all with different messages.